utils/ccdatofhir/mapping_resource_condition.bal

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// --------------------------------------------------------------------------------------------#
+// Source C-CDA to FHIR - Resource Mappings
+// --------------------------------------------------------------------------------------------#
+
+import ballerinax/health.fhir.r4;
+
+# Map CCDA Problem Observation to FHIR Condition.
+#
+# + actElement - actElement for CCDA Problem Observation
+# + return - FHIR Condition
+public isolated function mapCcdaConditionToFhir(xml actElement) returns r4:Condition? {
+    if isXMLElementNotNull(actElement) {
+        r4:Condition condition = {subject: {}};
+
+        xml idElement = actElement/<v3:id|id>;
+        xml effectiveTimeLowElement = actElement/<v3:effectiveTime|effectiveTime>/<v3:low|low>;
+        xml effectiveTimeHighElement = actElement/<v3:effectiveTime|effectiveTime>/<v3:high|high>;
+        xml valueElement = actElement/<v3:value|value>;
+        xml authorElement = actElement/<v3:author|author>;
+        xml observationValueElement = actElement/<v3:entryRelationship|entryRelationship>/<v3:observation|observation>/<v3:value|value>;
+
+        int index = 0;
+        foreach xml idElem in idElement {
+            r4:Identifier? mapCcdaIdToFhirIdentifierResult = mapCcdaIdToFhirIdentifier(idElem);
+            if mapCcdaIdToFhirIdentifierResult is r4:Identifier {
+                condition.identifier[index] = mapCcdaIdToFhirIdentifierResult;
+                index = index + 1;
```

Comment:
you can use `condition.identifier.push(mapCcdaIdToFhirIdentifierResult);`

Reply:
or you can replace the whole foreach statement with a query expression to reduce the line count

Reply:
Updated in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/e973cad61936b398a3651cbc34f89d9f66c2aa15